src/ApiClient.js

```diff
@@ -494,15 +494,32 @@ export default class ApiClient {
             }
         }
 
+        return this.sendApiRequest(request, returnType)
+    }
+
+    /**
+     * Sends the generated superagent request and deserializes the response.
+     * @param {module:superagent.Request} request The superagent request to send.
+     * @param {String} returnType The type to deserialize the response into.
+     * @returns {Promise} A {@link https://www.promisejs.org/|Promise} object.
+     */
+    sendApiRequest(request, returnType) {
         return new Promise((resolve, reject) => {
             request.end((error, response) => {
                 if (error) {
 
                     // Looks like there was an fault returned from the API
                     const hasErrorMessage = error.response && error.response.text
                     if (hasErrorMessage) {
-                        const fault = Fault.constructFromObject(JSON.parse(error.response.text).fault)
-                        reject(fault)
+                        try {
+                            const fault = Fault.constructFromObject(JSON.parse(error.response.text).fault)
+                            reject(fault)
+                        } catch (e) {
```

Comment:
Can you change `e` to `err`. Just a consistency thing with the rest of the library.